aws/resource_aws_cur_report_definition.go

```diff
@@ -0,0 +1,189 @@
+
+	reportName := *aws.String(d.Get("report_name").(string))
+
+	reportDefinition := &costandusagereportservice.ReportDefinition{
+		ReportName:  &reportName,
+		TimeUnit:    aws.String(d.Get("time_unit").(string)),
+		Format:      aws.String(d.Get("format").(string)),
+		Compression: aws.String(d.Get("compression").(string)),
+		S3Bucket:    aws.String(d.Get("s3_bucket").(string)),
+		S3Prefix:    aws.String(d.Get("s3_prefix").(string)),
+		S3Region:    aws.String(d.Get("s3_region").(string)),
+	}
+
+	additionalSchemaElementsSet := d.Get("additional_schema_elements").(*schema.Set)
+	for _, additionalSchemaElement := range additionalSchemaElementsSet.List() {
+		reportDefinition.AdditionalSchemaElements = append(reportDefinition.AdditionalSchemaElements,
+			aws.String(additionalSchemaElement.(string)))
+	}
+
+	additionalArtifactsSet := d.Get("additional_artifacts").(*schema.Set)
+	for _, additionalArtifact := range additionalArtifactsSet.List() {
+		reportDefinition.AdditionalArtifacts = append(reportDefinition.AdditionalArtifacts,
+			aws.String(additionalArtifact.(string)))
+	}
+
+	reportDefinitionInput := &costandusagereportservice.PutReportDefinitionInput{
+		ReportDefinition: reportDefinition,
+	}
+	log.Printf("[DEBUG] Creating AWS Cost and Usage Report Definition : %v", reportDefinitionInput)
+
+	_, err := conn.PutReportDefinition(reportDefinitionInput)
+	if err != nil {
+		return fmt.Errorf("Error creating AWS Cost And Usage Report Definition: %s", err)
+	}
+	d.SetId(reportName)
+	return resourceAwsCurReportDefinitionRead(d, meta)
+}
+
+func resourceAwsCurReportDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).costandusagereportconn
+
+	reportName := *aws.String(d.Id())
+
+	params := &costandusagereportservice.DescribeReportDefinitionsInput{}
+
+	log.Printf("[DEBUG] Reading CurReportDefinition: %s", reportName)
+
+	var matchingReportDefinition *costandusagereportservice.ReportDefinition
+	err := conn.DescribeReportDefinitionsPages(params, func(resp *costandusagereportservice.DescribeReportDefinitionsOutput, isLast bool) bool {
+		for _, reportDefinition := range resp.ReportDefinitions {
+			if *reportDefinition.ReportName == reportName {
+				matchingReportDefinition = reportDefinition
+				return false
+			}
+		}
+		return !isLast
+	})
+	if err != nil {
+		return err
+	}
+	if matchingReportDefinition == nil {
+		log.Printf("[WARN] Report definition (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	} else {
+		d.SetId(*matchingReportDefinition.ReportName)
+		d.Set("report_name", matchingReportDefinition.ReportName)
+		d.Set("time_unit", matchingReportDefinition.TimeUnit)
+		d.Set("format", matchingReportDefinition.Format)
+		d.Set("compression", matchingReportDefinition.Compression)
+		d.Set("additional_schema_elements", aws.StringValueSlice(matchingReportDefinition.AdditionalSchemaElements))
+		d.Set("s3_bucket", *matchingReportDefinition.S3Bucket)
```

Comment:
To prevent potential panics here and below, we should either remove the `*` dereference or replace it with a wrapping `aws.StringValue(matchingReportDefinition.XXX)`